ddl/generated_column.go

```diff
@@ -198,6 +207,34 @@ func checkIllegalFn4GeneratedColumn(colName string, expr ast.ExprNode) error {
 	return nil
 }
 
+// Check whether newColumnDef refers to any auto-increment columns.
+func checkGeneratedWithAutoInc(tableInfo *model.TableInfo, newColumnDef *ast.ColumnDef) error {
+	_, dependColNames := findDependedColumnNames(newColumnDef)
+	if err := checkAutoIncrementRef(newColumnDef.Name.Name.L, dependColNames, tableInfo); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func checkIndexOrStored(tbl table.Table, oldCol, newCol *table.Column) error {
+	if oldCol.GeneratedExprString == newCol.GeneratedExprString {
+		return nil
+	}
+
+	if newCol.GeneratedStored {
+		return errUnsupportedOnGeneratedColumn.GenWithStackByArgs("modifying a stored column")
+	}
+
+	for _, idx := range tbl.Indices() {
+		for _, col := range idx.Meta().Columns {
+			if col.Name.L == newCol.Name.L {
```

Comment:
why not check `col.Name.L == oldCol.Name.L`?

Reply:
@winkyao If the new column is NOT stored or index, then no backfilling is needed. So there is nothing to do with the old column.